### PR TITLE
Various fixes for issue mail notifications

### LINKED ISSUE #5977: Issue closure with empty comment does not generate E-Mail
Gitea version (or commit ref): https://github.com/go-gitea/gitea/commit/0c721fe016919e368e105142d45efcd4d477757b

When a issue is closed with a empty comment box, no E-Mail seems to be sent to watchers.

### LINKED ISSUE #7124: Issue closure email has wrong `From` header
To reproduce:

1. Open issue as `user1`
2. Close issue as `user2`
3. The resulting mail with body `Closed #id` is using `From: user1` header, should be `From: user2`.

### PR DESCRIPTION
**Pull request by silverwind**
Various fixes for issue mail notifications

- Send individual mails for actions and comments
- Send mail for new issues/prs without a comment
- Use correct sender for reopen/close actions
- Hopefully fixed all bugs related to missing mails

Fixes: https://github.com/go-gitea/gitea/issues/7124
Fixes: https://github.com/go-gitea/gitea/issues/5977

### CODE REVIEW

File: models/issue_mail.go [hunk 1]
```diff
@@ -118,26 +118,41 @@ func mailIssueCommentToParticipants(e Engine, issue *Issue, doer *User, content
...
+
 	if err = UpdateIssueMentions(e, issue.ID, mentions); err != nil {
 		return fmt.Errorf("UpdateIssueMentions [%d]: %v", issue.ID, err)
 	}
 
-	var content = issue.Content
+	if len(issue.Content) > 0 {
+		if err = mailIssueCommentToParticipants(e, issue, doer, issue.Content, nil, mentions); err != nil {
+			log.Error("mailIssueCommentToParticipants: %v", err)
+		}
```
Comment (lunny) [inline models/issue_mail.go:135]: return?

File: models/issue_comment.go [hunk 2]
```diff
@@ -403,16 +403,23 @@ func (c *Comment) mailParticipants(e Engine, opType ActionType, issue *Issue) (e
 		return fmt.Errorf("UpdateIssueMentions [%d]: %v", c.IssueID, err)
 	}
 
-	content := c.Content
+	if len(c.Content) > 0 {
+		if err = mailIssueCommentToParticipants(e, issue, c.Poster, c.Content, c, mentions); err != nil {
+			log.Error("mailIssueCommentToParticipants: %v", err)
+		}
```
Comment (lunny) [inline models/issue_comment.go:409]: return?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/issue_comment.go (hunk 2 above)
Comment (silverwind) [reply models/issue_comment.go:409]: No, sending two mails is intended. One for each action (comment and close/reopen).

File: models/issue_mail.go (hunk 1 above)
Comment (silverwind) [reply models/issue_mail.go:135]: Same here, it's intended.

File: models/issue_comment.go (hunk 2 above)
Comment (silverwind) [reply models/issue_comment.go:409]: If it's just a comment, the `switch` below will not hit a case and only 1 mail is sent.

Comment (lunny) [reply models/issue_comment.go:409]: One action sending two mails is annoying.

Comment (silverwind) [reply models/issue_comment.go:409]: I don't think so. It's not one action after all if a user clicks "comment and close". Also it does match GitHub's behaviour, they also send two mails on "comment and close".

Comment (silverwind) [reply models/issue_comment.go:409]: Example of GitHub sending two mails for "comment and close":

image

**Review by lunny — APPROVED** [review]

**Review by mrsdizzie — APPROVED** [review]

**Pull request merged**